src/cockpit/channels/stream.py

```diff
@@ -86,7 +86,7 @@ def do_close(self):
             # already gone? fine!
             logger.debug('Received close signal from peer, but process %i is already gone', pid)
 
-    def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, object]) -> SubprocessTransport:
+    async def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, object]) -> SubprocessTransport:
```

Comment:
...and maybe this one too, although it could be more work.

Reply:
[subprocess.exec()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.subprocess_exec) looks nice, but it doesn't support the `pty`/`windows` options. Perhaps this can be injected or subclassed, but this smells more like a follow-up?